src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -1,9 +1,135 @@
+        return !isArmInInterferenceZone();
+    }
+
+    public void rezeroArm() {
+        armMotorLeft.setPosition(armLefttEncoderAbs.getAbsolutePosition().getValueAsDouble() * 360);
+    }
+
+    // Account for PID when setting position of our arm
+    public void controlPosition(double inputPositionDegrees) {
+        PositionVoltage desiredVoltage = new PositionVoltage(inputPositionDegrees / 360).withSlot(0);
+        armMotorRight.setControl(desiredVoltage);
+        // desired pos = requested pos
+        this.desiredSetpointPosition = inputPositionDegrees;
+        // convert velocity in rotations/sec to deg/sec
+        this.desiredSetpointVelocity = armMotorRight.getVelocity().getValueAsDouble() * 360;
+    }
+
+    public boolean atPositionDegrees(ArmPosition pos) {
+        double checkPositionDeg = armPositions.get(pos);
+        double currentPositionDeg = armMotorLeft.getPosition().getValueAsDouble() * 360;
+
+        return Math.abs(checkPositionDeg - currentPositionDeg) < ArmCal.ARM_MARGIN_DEGREES;
+    }
+    public boolean isArmInInterferenceZone() {
+        double currentPosition = armMotorLeft.getPosition().getValueAsDouble() * 360;
+        if (currentPosition <= ArmCal.ARM_INTERFERENCE_THRESHOLD_MAX_DEGREES && 
+            currentPosition >= ArmCal.ARM_INTERFERENCE_THRESHOLD_MIN_DEGREES 
+            /** TODO: AND is Deep Climb engaged */ ) {
+            return true;
+        }
+        return false;
+    }
     
-}
+
+    public void stopArmMovement() {
+        // left motor follows right motor, so armMotorLeft is not necessary here
+        armMotorLeft.setVoltage(0.0);
+    }
+
+    @Override
+    public void periodic() {
+        // control arm position
+        if (isArmMoveable()) {
+            controlPosition(armPositions.get(this.armDesiredPosition));
+        }
+    }
+    @Override
+    public void initSendable(SendableBuilder builder) {
+        super.initSendable(builder);
+        builder.addDoubleProperty("Desired Setpoint Position (Deg)", (() -> desiredSetpointPosition), null);
+        builder.addDoubleProperty("Current Position (Relative Encoder) (Deg)", (() -> armMotorRight.getPosition().getValueAsDouble() * 360), null);
+        builder.addDoubleProperty("Desired Setpoint Velocity (m/s)", (() -> desiredSetpointVelocity), null);
+        builder.addDoubleProperty("Arm Demand Volts (calc)", (() -> armDemandVolts), null);
+        builder.addBooleanProperty("Is Arm In Interference Zone", this::isArmInInterferenceZone, null);
+        builder.addDoubleProperty("Motor Right Angle (Relative) ", (() -> armMotorRight.getPosition().getValueAsDouble() * 360), null);
+        builder.addDoubleProperty("Motor Left Angle (Relative) ", (() -> armMotorLeft.getPosition().getValueAsDouble() * 360), null);
+        builder.addDoubleProperty("Motor Right Angle (Absolute) ", (() -> armLefttEncoderAbs.getAbsolutePosition().getValueAsDouble() * 360), null);
```

Comment:
say "right motor angle" not "motor right angle" because this makes it sound like it's at 90 degrees*mol/kelvin per joule

Reply:
oh, and it's 360.0